zarr_checksum/generators.py

```diff
@@ -73,32 +72,22 @@ def yield_files_s3(
         yield from []
         return
 
-    # Iterate until all files found
-    while True:
-        if continuation_token is not None:
-            options["ContinuationToken"] = continuation_token
-
-        # Fetch
-        res = client.list_objects_v2(**options)
-
+    # Fetch
+    paginator = client.get_paginator("list_objects_v2")
+    for page in paginator.paginate(**options):
```

Comment:
```suggestion
    for page in paginator.paginate(Bucket=bucket, Prefix=prefix):
```

There's no need for an `options` dict anymore, and using `**kwargs` hurts type-checking.